### PR TITLE
Docker + docker-compose

### PR DESCRIPTION
**Pull request by imobachgs**
The first version of a Docker manifest and docker-compose files. To be used in development only!

### CODE REVIEW

File: docker/database.mysql.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SQLite version 3.x
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem 'sqlite3'
+development:
+  adapter: mysql2
+  database: tsp_development
+  host: db
+  username: root
```
Comment (cboltz) [inline docker/database.mysql.yml]: Even if this is only an example config, I'd recommend to have a secure example. `Username: root` does not qualify as such ;-)

The same applies for Postgresql - please also use a non-privileged user in the example config there.

File: docker/docker-compose.mysql.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+version: '3'
+services:
+  web:
+    image: tsp:latest
+    build: .
+    volumes:
+      - .:/app
+      - /tmp/.X11-unix:/tmp/.X11-unix
+      - ~/.Xauthority:/root/.Xauthority
```
Comment (cboltz) [inline docker/docker-compose.mysql.yml]: I'm not a docker or ruby expert, but mounting `/tmp/.X11-unix` and `Xauthority` into the container looks strange for a webapp.

Is there a special reason why you do this?

File: docker/database.mysql.yml (hunk 1 above)
Comment (imobachgs) [reply docker/database.mysql.yml]: This is just a development-oriented example. In such a case, you need to have privileges to create/destroy databases. However, I can try to come up with a more secure alternative.

File: docker/docker-compose.mysql.yml (hunk 2 above)
Comment (imobachgs) [reply docker/docker-compose.mysql.yml]: Yes, there is a reason: running the integration tests using a browser. In this case, you need to allow `capybara-webkit` to connect to your X server. As said before, this configuration is oriented for development only :)

Of course, when deploying the application, you shouldn't do that.

Comment (imobachgs) [reply docker/docker-compose.mysql.yml]: I will try to make more clear that this docker-compose thingie is just for development and I will provide a production oriented docker manifest.

File: docker/database.mysql.yml (hunk 1 above)
Comment (cboltz) [reply docker/database.mysql.yml]: I don't even want to know how many "development-orientated examples" of various software are running in production... ;-)

Proposal: change it to something like
```
# needed MySQL permissions: full permissions (including create/drop database) for the database given above
- username: tsp-dev
```

File: docker/docker-compose.mysql.yml (hunk 2 above)
Comment (cboltz) [reply docker/docker-compose.mysql.yml]: Thanks for the explanation.

As an idea - renaming this file to something like `docker-compose.development.mysql.yml` would be a good start ;-)

**Comment by ancorgs** [comment]
Closed in favor of #106

**Pull request closed without merging**